pkg/discovery/dns/provider.go

```diff
@@ -53,11 +53,20 @@ func (t ResolverType) ToResolver(logger log.Logger) ipLookupResolver {
 	return r
 }
 
+// Deprecated. Use NewProviderWithReturnOnErrorIfNotFound instead.
+//
 // NewProvider returns a new empty provider with a given resolver type.
 // If empty resolver type is net.DefaultResolver.
+// TODO(OGKevin): Refactor code to use new method and eventually rename NewProviderWithReturnOnErrorIfNotFound back to NewProvider.
 func NewProvider(logger log.Logger, reg prometheus.Registerer, resolverType ResolverType) *Provider {
+	return NewProviderWithReturnOnErrorIfNotFound(logger, reg, resolverType, true)
```

Comment:
I am pretty sure there are better ways to handle this then this. Something not involving super long method name and boolean flag (: